### PR TITLE
Speed up sprite generation and disable colormap by default

### PR DESCRIPTION
**Pull request by toddr**

### CODE REVIEW

**Comment by kaoru** [comment]
Thanks @toddr!

Am I reading right that there are three main changes here:

1. Expose an option to disable color mapping entirely
2. Don't re-color map unnecessarily (`return 1 if ref  $rh_info->{colors}{map};`)
3. Optimize `_write_image` using `Image::Magick::Composite`

Is (1) still needed given (2)? I'll be honest I didn't write the color mapping stuff so I'm not sure what it's for 😅

**Comment by bdraco** [comment]
Hi @kaoru 
The enable_colormap option was added for callers who are relying on preserving the original (slow) color map being generated.  If its not being used it could just be ripped out but that might cause pain for other consumers using the module.

File: lib/CSS/SpriteMaker.pm [hunk 1]
```diff
@@ -1430,31 +1415,14 @@ sub _get_image_properties {
         $rh_info->{height} = $first_bottom - $first_top + 1;
     }
 
+    if ($enable_colormap) {
+      $self->_generate_color_map_for_image_properties($Image, $rh_info);
```
Comment (darksmo) [inline lib/CSS/SpriteMaker.pm]: Typo, should be _generate_**colormap**_for_image_properties

File: lib/CSS/SpriteMaker.pm [hunk 2]
```diff
@@ -548,6 +554,7 @@ EONCLICK
             next if $key eq "colors";
             print $fh "<b>" . $key . "</b>: " . ($rh_source_info->{$key} // 'none') . "<br />";
         }
+        if (!$self->{enable_colormap}) {
```
Comment (darksmo) [inline lib/CSS/SpriteMaker.pm]: Shouldn't this be if ($self->{enable_colormap}) ?

File: lib/CSS/SpriteMaker.pm [hunk 3]
```diff
@@ -548,6 +554,7 @@ EONCLICK
...
         print $fh '<h3>Colors</h3>';
             print $fh "<b>total</b>: " . $rh_source_info->{colors}{total} . '<br />';
```
Comment (darksmo) [inline lib/CSS/SpriteMaker.pm]: I would still print this, as the total number of colors is not determined based on the enable_colormap option

**Comment by darksmo** [comment]
So I merged this, fixed the above, wrote some tests... thinking about releasing version 1.01

**Comment by kaoru** [comment]
Nice, thanks Savio!

**Pull request merged**